### PR TITLE
Proposal: API update

### PR DESCRIPTION
**Pull request by edgardz**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [x] Code style update
- [x] Refactor (refactoring or adding test which isn't a fix or add a feature)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes
- [ ] No

**Did you test your solution?**

- [ ] I lightly tested it in one browser
- [x] I deeply tested it in several browsers
- [ ] I wrote tests around it (unit tests, integration tests, E2E tests)

## Problem Description

The current detect lib is not updated for 2 years and was lacking some new things we might need nowadays.

## Solution Description

This PR introduces a new and cleaner detect API with new features like:

- SSR fallback.
- Spoofed iPad detection.
- Improved in-app detection, including WeChat and TikTok.
- Improved bot detection.

## Side Effects, Risks, Impact

Since the API changed, it is no longer compatible with the 1.x version implementation.

**Aditional comments:**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,17 +1,163 @@
...
+    const w = Math.max(document.documentElement.clientWidth, detector.window.innerWidth || 0);
+    const h = Math.max(document.documentElement.clientHeight, detector.window.innerHeight || 0);
+    return w < h ? 'PORTRAIT' : 'LANDSCAPE';
+  }
+  get portrait() {
+    return this.orientation === 'PORTRAIT';
+  }
+  get landscape() {
+    return this.orientation === 'LANDSCAPE';
+  }
```
Comment (DonghyukJacobJang) [inline src/index.js:157]: @edgardz  we are currently relying on the `useOrientation` hook from `react-use` library. Do you have any other good potential use cases if we have this in the `detect` library?

**Comment by DonghyukJacobJang** [comment]
Thanks for this PR. There are great updates that we have been thinking of; iPad (ios13), more in-app browser types, and more. 

I am thinking we may have to convert this to Typescript. What do you think?

**Comment by edgardz** [comment]
I have a Typescript version of it. Should I update now?

**Comment by DonghyukJacobJang** [comment]
@edgardz yes, please add TS version so that we can use all advantages of it.

**Comment by DonghyukJacobJang** [comment]
I got an error below when running `npm run test`
Screen Shot 2021-08-26 at 2 01 53 PM

The error was solved by installing `tslib`. It might be installed in your machine globally. So, let's add this in dependency list.

Once it is done, we can merge 🚀

**Pull request merged**